065-topic-replication-factor-change.md

```diff
@@ -0,0 +1,308 @@
+They will be able to change the topic replication factor by simply updating the KafkaTopic resource, which may also happen through a GitOps pipeline.
+
+This is a non exhaustive list of use cases that will become significantly easier:
+
+- Set a lower replication factor for non-critical topics to reduce the infrastructure costs (compute, storage and network).
+- Set a lower replication factor to deal with resource shortage (trade-off between fault-tolerance and resource utilization).
+- Set a higher replication factor for critical topics to improve durability and fault tolerance.
+
+A tighter integration with Cruise Control is also one of the long term goals of the Strimzi project. 
+
+## Proposal
+
+The replication factor change feature can be implemented by integrating the Topic Operator with Cruise Control.
+The Cluster Operator is already integrated with Cruise Control, so we can factor out the existing logic and classes.
+
+Cruise Control creates a task for each operation, such as topic configuration and cluster rebalance.
+A task is first created in the Active state, then moves to InExecution, and finally to Completed or CompletedWithError.
+At most `max.active.user.tasks` tasks can be Active at the same time (5 by default), and only one task can be InExecution at any given time.
+
+During the execution, a task is divided in one or more batches, which are then executed one after the other.
+If any batch fails then the task becomes `CompletedWithError` potentially leaving some topics in a changed state and others unchanged.
+A completed task is retained for `completed.user.task.retention.time.ms`, which defaults to 24 hours.
+
+### Topic configuration
+
+The [`topic_configuration`](https://github.com/linkedin/cruise-control/wiki/REST-APIs#change-kafka-topic-configuration) endpoint will be used to request replication factor changes.
+Once accepted, the replication factor change is put into the Cruise Control's task queue and executed asynchronously with respect to the sending of the HTTP response to the request that initiated the change.
+
+The two required parameters for this endpoint are `topic` (regular expression to specify subject topics) and `replication_factor` (target replication factor).
+In order to group multiple changes in the same request to Cruise Control, the Topic Operator uses a [JSON payload](https://github.com/linkedin/cruise-control/wiki/Change-topic-replication-factor-through-Cruise-Control#instruction), which replaces the required parameters.
+This JSON contains a list of replication factor values, each one associated with a regex expression of the form `topic1|topic2|topic3` matching all topics with that target replication factor.
+
+This is useful with bulk replicas change, which can happen in some edge cases.
+A potentially big backlog of changes may be created when the Topic Operator is not running, or there is an issue with Cruise Control or Kafka.
```

Comment:
If that's what we want to say, is it clear that this a potential drawback when making bulk changes? Or are we saying that the infrastructure must be capable of handling the bulk changes?

Reply:
The infrastructure must be capable of handling the bulk changes.